smallfile-wrapper/smallfile-wrapper.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import elasticsearch
+import os
+import sys
+import json
+import traceback
+
+
+# The purpose of this function is to consolidate all result jsons in to 1 list
+def _load_result_JSON(smallfile_path, operations):
+    results = []
+    for operation in operations:
+        # smallfile results are stored in directory, {{smallfile_path}}/RESULTS
+        # naming convention for result JSON is: SMALLFILE_$operation_RUN_RESULT
+        _path = smallfile_path+"/RESULTS/SMALLFILE_"+operation+"_RUN_RESULT"
+        with open(_path) as json_file:
+            data = json.load(json_file)
+            results.append(data)
+    return results
+
+
+# The purpose of this function is to establish connection with
+# elasticsearch server and index the consolidated results 1 by 1
+def _index_result(server, port, payload):
+    index = "smallfile-results"
+    try:
+        es = elasticsearch.Elasticsearch([
+            {'host': server, 'port': port}], send_get_body_as='POST')
+        for result in payload:
+            es.index(index=index, doc_type="result", body=result)
+    except Exception as e:
+        sys.stdout.write("******Exception Occured*******")
+        sys.stdout.write(str(e))
+        sys.stdout.write(traceback.format_exc())
+        return False
+
+
+# The pupose of this function is to inject the uuid and user into
+# the data generated by smallfile.
+def _json_payload(data, uuid, user):
+    processed = []
+    for item in data:
+        item["workload"] = "smallfile"
+        item["uuid"] = uuid
+        item["user"] = user
+        processed.append(item)
+    return processed
+
+
+def main():
+
+    # retrieving elasticsearch server address and operating port from
+    # environement variables. The credentials are exported in the
+    # tasks of ripsaw/roles/smallfile-bench. These variables are exported
+    # based on condition, if the es_port and es_server is defined in
+    # smallfile CRDS of ripsaw. Use of "os.environ.get($env_var)" is to look
+    # for the supplied $env_var. If there is no such $env_var defined,
+    # then it will return None object.
+    server = os.environ.get("es")
```

Comment:
we should  check to see if these exists prior to capturing them.

Reply:
I deliberately used os.environ.get("es"). Evenif, es doesn't exist, this will simply return None and i am checking for the None object later in the code.